src/tasks/task_list.hpp

```diff
@@ -110,10 +102,39 @@ class TaskList {
  protected:
   std::list<Task> task_list_;
   int tasks_added_ = 0;
-  std::vector<TaskList *> dependencies_;
   TaskID tasks_completed_;
 };
 
+using TaskRegion = std::vector<TaskList>;
+
+struct TaskCollection {
+  TaskCollection() = default;
+  TaskRegion &AddRegion(const int num_lists) {
+    regions.push_back(TaskRegion(num_lists));
+    return regions.back();
+  }
+  TaskListStatus Execute() {
+    for (auto region : regions) {
```

Comment:
Is there a reason this cannot be a reference?

Reply:
don't think so...fixed.